src/Microsoft.Extensions.Caching.Memory/MemoryCache.cs

```diff
@@ -302,6 +258,68 @@ private void EntryExpired(CacheEntry entry)
             StartScanForExpiredItems();
         }
 
+        private void SetEntry(CacheEntry entry)
+        {
+            var utcNow = _clock.UtcNow;
+
+            DateTimeOffset? absoluteExpiration = null;
+            if (entry._absoluteExpirationRelativeToNow.HasValue)
+            {
+                absoluteExpiration = utcNow + entry._absoluteExpirationRelativeToNow;
+            }
+            else if (entry._absoluteExpiration.HasValue)
+            {
+                if (entry._absoluteExpiration <= utcNow)
+                {
+                    throw new ArgumentOutOfRangeException(
```

Comment:
We have a unit test that ensures an exception is thrown when adding an absolute expiration in the past, this case. Dispose actually throws in this case and fulfills the unit test as the SetEntry defines the utcNow to take into account, so I would preferably left it here. We can test the value when it's set as the current time is not valid at that moment.